app/controllers/targets_controller.rb

```diff
@@ -0,0 +1,15 @@
+class TargetsController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    current_user.targets.create!(target_params)
+  rescue ActiveRecord::RecordInvalid => e
```

Comment:
Maybe you can include this on ApplicationController instead using something like `rescue_from`